astra/bundle.go

```diff
@@ -85,6 +94,68 @@ func LoadBundleZipFromPath(path string) (*Bundle, error) {
 	return LoadBundleZip(&reader.Reader)
 }
 
+func LoadBundleZipFromURL(url, databaseID, token string, timeout time.Duration) (*Bundle, error) {
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
+	defer cancel()
+
+	credsURL, err := generateSecureBundleURLWithResponse(url, databaseID, token, ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error generating secure bundle zip URLs: %v", err)
+	}
+	resp, err := http.Get(credsURL.DownloadURL)
+
+	defer resp.Body.Close()
+
+	body, err := readAllWithTimeout(resp.Body, ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error downloading bundle zip: %v", err)
+	}
+
+	reader, err := zip.NewReader(bytes.NewReader(body), int64(len(body)))
+	if err != nil {
+		return nil, fmt.Errorf("error creating zip reader for bundle zip: %v", err)
+	}
+
+	return LoadBundleZip(reader)
+}
+
+func readAllWithTimeout(r io.Reader, ctx context.Context) (bytes []byte, err error) {
+	ch := make(chan struct{})
+
+	go func() {
+		bytes, err = ioutil.ReadAll(r)
```

Comment:
The error needs to be returned. Make `ch` an `error` channel and return that way.